### PR TITLE
chore: Add code formatting

### PR DESCRIPTION
**Pull request by jbuget**
## :wrench: Problem

There is no formatting rules clearly defined.
It makes it hard to contribute.
Some code changes can be avoid, making code review easier.

## :cake: Solution

Add [Prettier](https://prettier.io/) and [`.editorconfig` file](https://editorconfig.org/).

Add 2 npm dev tasks : 
- `format:check`
- `format:write`

Include `format:check` to CI "Test" job.

## :rotating_light:  Points of interest

> _If any, details what are the edge cases, or any additional topics to share about._

## :desert_island: How to test

Check that the 2 formatting tasks work.
Check that checks are taken into account by GHA.

based on #4 from @flemzord (thanks ❤️ ❤️ ❤️)

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 name: Main
 jobs:
   Test:
-    name: 'Test'
+    name: 'Lint & test'
```
Comment (flemzord) [inline .github/workflows/main.yaml]: I think it would be better to separate it into 2 different jobs.

**Review by flemzord — APPROVED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (jbuget) [reply .github/workflows/main.yaml]: I hesitated. Ok. I do it.
